composer/lib/dependabot/composer/update_checker/version_resolver.rb

```diff
@@ -72,9 +72,10 @@ def fetch_latest_resolvable_version
         def fetch_latest_resolvable_version_string
           base_directory = dependency_files.first.directory
           SharedHelpers.in_a_temporary_directory(base_directory) do
-            File.write("composer.json", prepared_composer_json_content)
-            File.write("composer.lock", lockfile.content) if lockfile
-            File.write("auth.json", auth_json.content) if auth_json
+            write_dependency_file
+            write_path_dependency_files
+            write_lockfile
+            write_auth_file
```

Comment:
Looks like we also write out the files in `check_original_requirements_resolvable`